experimental/LieAlgebras/src/RootSystem.jl

```diff
@@ -638,36 +671,55 @@ function height(r::DualRootSpaceElem)
   return sum(coefficients(r))
 end
 
+function is_coroot(r::DualRootSpaceElem)
+  return is_positive_coroot(r) || is_negative_coroot(r)
+end
+
 function is_coroot_with_index(r::DualRootSpaceElem)
-  i = findfirst(==(r), coroots(root_system(r)))
-  if isnothing(i)
-    return false, 0
-  else
+  fl, i = is_positive_coroot_with_index(r)
+  if fl
     return true, i
   end
+  fl, j = is_negative_coroot_with_index(r)
+  if fl
+    return true, j + number_of_positive_roots(root_system(r))
+  end
+  return false, 0
+end
+
+function is_positive_coroot(r::DualRootSpaceElem)
+  return haskey(root_system(r).positive_coroots_map, coefficients(r))
 end
 
 function is_positive_coroot_with_index(r::DualRootSpaceElem)
-  i = findfirst(==(r), positive_coroots(root_system(r)))
+  i = get(root_system(r).positive_coroots_map, coefficients(r), nothing)
   if isnothing(i)
     return false, 0
   else
     return true, i
   end
```

Comment:
Obv. this is existing code, and it is fine, but just as food for thought:
```suggestion
  i = get(root_system(r).positive_coroots_map, coefficients(r), 0)
  return !is_zero(i), i
```

Reply:
will be included in the next round of refactoring :)